src/neighbors.cc

```diff
@@ -1141,15 +1142,15 @@ peerDNSConfigure(const ipcache_addrs *ia, const Dns::LookupDetails &, void *data
     p->in_addr = p->addresses[0];
     p->in_addr.port(p->icp.port);
 
-    peerProbeConnect(p, true); // detect any died or revived peers ASAP
+    peerProbeConnect(p.getRaw(), true); // detect any died or revived peers ASAP
 
     if (p->type == PEER_MULTICAST)
-        peerCountMcastPeersSchedule(p, 10);
+        peerCountMcastPeersSchedule(p.getRaw(), 10);
 
 #if USE_ICMP
     if (p->type != PEER_MULTICAST && IamWorkerProcess())
         if (!p->options.no_netdb_exchange)
-            eventAddIsh("netdbExchangeStart", netdbExchangeStart, p, 30.0, 1);
+            eventAddIsh("netdbExchangeStart", netdbExchangeStart, p.getRaw(), 30.0, 1);
```

Comment:
eventAddIsh() implies eventAdd() with true cbdata parameter, but `p.getRaw()` (i.e. a CachePeer pointer) is not a pointer to a cbdata-protected class!

Reply:
If my concern is valid _and_ ipcache_nbgethostbyname() enhancement requested in a sibling issue is successful, then we might be able to adjust event.h APIs to catch such bugs automatically.